src/main/java/fi/otavanopisto/kuntaapi/server/integrations/casem/CaseMCacheUpdater.java

```diff
@@ -0,0 +1,839 @@
+      NodeList nodeList = getChildNodeList(organizationId, caseMRootNodeId, caseMParentIds, skipToken);
+      if (nodeList == null) {
+        break;
+      }
+      
+      result.addAll(nodeList.getValue());
+      
+      skipToken = getSkipToken(nodeList.getOdataNextLink());
+    } while (skipToken != null);
+    
+    Collections.sort(result, new NodeComparator());
+    
+    return result;
+  }
+  
+  private NodeList getChildNodeList(OrganizationId organizationId, Long caseMRootNodeId, List<Long> caseMParentIds, Long skipToken) {
+    String pathQuery = getSubNodePath(caseMParentIds);
+    
+    ApiResponse<NodeList> response = caseMApi.getNodesApi(organizationId)
+      .listSubNodes(caseMRootNodeId, pathQuery, skipToken != null ? String.valueOf(skipToken) : null);
+    
+    if (!response.isOk()) {
+      logger.severe(String.format("Listing nodes by rootNode %d and pathQuery %s failed on [%d] %s", caseMRootNodeId, pathQuery, response.getStatus(), response.getMessage()));
+      return null;
+    } else {
+      return response.getResponse();
+    }
+  }
+  
+  private String getSubNodePath(List<Long> caseMParentIds) {
+    StringBuilder result = new StringBuilder();
+    
+    for (Long caseMParentId : caseMParentIds) {
+      result.append(String.format("SubNodes(%d)/", caseMParentId));
+    }
+    
+    result.append("SubNodes()");
+    
+    return result.toString();
+  }
+  
+  private Long getSkipToken(String nextLink) {
+    if (StringUtils.isBlank(nextLink)) {
+      return null;
+    }
+    
+    Pattern pattern = Pattern.compile("(.*\\$skiptoken=)([0-9]*)");
+    Matcher matcher = pattern.matcher(nextLink);
+    if (matcher.matches() && (matcher.groupCount() > 1)) {
+      return NumberUtils.createLong(matcher.group(2));
+    }
+    
+    return null;
+  }
+  
+  private Page translateNode(OrganizationId organizationId, Long caseMRootNodeId, Node parentNode, Node node) {
+    Page page = new Page();
+    PageId kuntaApiId = translatePageId(toNodeId(organizationId, node.getNodeId()), true);
+    PageId kuntaApiParentId = null;
+    
+    if (parentNode != null && !parentNode.getNodeId().equals(caseMRootNodeId)) {
+      if (parentNode.getType().equals(0l)) {
+        kuntaApiParentId = parentNode.getParentId() != null ? translatePageId(toNodeId(organizationId, parentNode.getParentId()), false) : null;
+      } else {
+        kuntaApiParentId = translatePageId(toNodeId(organizationId, parentNode.getNodeId()), false);
+      }
+    }
+    
+    List<LocalizedValue> titles = translateNodeNames(organizationId, node.getNames());
+    page.setId(kuntaApiId.getId());
+    page.setParentId(kuntaApiParentId != null ? kuntaApiParentId.getId() : null);
+    page.setTitles(titles);
+    page.setSlug(slugify(titles.isEmpty() ? kuntaApiId.getId() : titles.get(0).getValue()));
+    
+    return page;
+  }
+  
+  private Page translateContent(OrganizationId organizationId, Page parentPage, Content content, String title, String slug) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Remove this unused private "translateContent" method. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AUnusedPrivateMethod)